x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -133,79 +104,44 @@ export const EditConnector = React.memo(
     const { actions } = useApplicationCapabilities();
     const actionsReadCapabilities = actions.read;
 
-    // by default save if disabled
-    const [enableSave, setEnableSave] = useState(false);
-
     const { setFieldValue, submit } = form;
 
     const [{ currentConnector, fields, editConnector }, dispatch] = useReducer(
       editConnectorReducer,
-      { ...initialState, fields: caseFields }
+      {
+        ...initialState,
+        fields: caseFields,
+        currentConnector: getConnectorById(caseData.connector.id, allAvailableConnectors),
```

Comment:
We init the state with data from the `caseData`. The use of `key` in `https://github.com/elastic/kibana/pull/149276/files?diff=split&w=1#r1083999611` resets the initial state of the reducer every time the `caseData.connector.id` changes.